### PR TITLE
NPC_MAGICMIRROR damage type

### PR DESCRIPTION
**Pull request by eppc0330**
* **Addressed Issue(s)**: #6709 

* **Server Mode**: Renewal

* **Description of Pull Request**: 
  - Now type of damage, reflected by NPC_MAGICMIRROR, will be special melee weapon damage.

  - Skill damage reflected by NPC_MAGICMIRROR will be able to trigger autospell.

  - Kyomu will block NPC_MAGICMIRROR reflecting ability.

  - Calculation of reducing reflect damage will be addition, not multiplication.

  Sorry I've deleted repository.. so I closed original pull and opened new one.

### CODE REVIEW

**Comment by CairoLee** [comment]
Previously, it was possible to determine whether damage was caused by a reflect skill by checking the skill ID for CR_REFLECTSHIELD. However, this will no longer be easily achievable after the applied changes.

Maybe we can add a flag to indicate that this is a reflect damage?

**Comment by eppc0330** [comment]
@CairoLee Sorry I am not following. More explain please? Why determining needed?

**Comment by CairoLee** [comment]
@eppc0330 Because that reflection damage should not trigger bAutoSpell (the accuracy of this behavior is only confirmed in twRO, I'm not sure about other versions).

I make this change in my own MOD. When this PR is merged, I may need to make bigger adjustments to the code, and it will also increase the probability of conflicts when I merge rathena's code into my own repository in the future. If you could consider this together, it would be better for me.

This is just a suggestion. Thanks again for your PR and attention.

**Comment by eppc0330** [comment]
@CairoLee Hmm.. so you are saying the opposite of mine. Because in KRO reflectied damage should be able to trigger autospell.

**Comment by eppc0330** [comment]
@CairoLee Oh okay I guess I got it. From your point of view, I think I was talking about autospellwhenhit. Current rA's reflect damage does not trigger autospellwhenhit.

**Comment by eppc0330** [comment]
@CairoLee Can you DM me? My discord id eppc0330.

**Comment by eppc0330** [comment]
@CairoLee Confirmed in KRo reflect damage not triggering autospellwhenhit.
Same in case of autobonus2, right?

**Comment by CairoLee** [comment]
@eppc0330 Yes, that's why I am requesting a maker to be provided, which can help me determine if the damage is caused by reflection.

This will allow me to skip the processing logic of autospell2 when necessary.

**Comment by eppc0330** [comment]
@CairoLee Thanks for nice feedback. I've reverted codes about reflectshield.
Now this pull is only doing with `NPC_MAGICMIRROR`.

File: src/map/battle.cpp [hunk 1]
```diff
@@ -8804,7 +8780,29 @@ int64 battle_calc_return_damage(struct block_list* tbl, struct block_list *src,
...
+	if (sc) {
+		if (status_reflect && sc->getSCE(SC_REFLECTDAMAGE)) {
+			reduce += sc->getSCE(SC_REFLECTDAMAGE)->val2;
+		}
+		if (sc->getSCE(SC_VENOMBLEED) && sc->getSCE(SC_VENOMBLEED)->val3 == 0) {
+			reduce += sc->getSCE(SC_VENOMBLEED)->val2;
+		}
+		if (sc->getSCE(SC_REF_T_POTION))
+			reduce += 100;
+	}
```
Comment (vstumpf) [inline src/map/battle.cpp:8801]: we can probably factor this out, but we'll do this in another pr

**Review by vstumpf — APPROVED** [review]
Sorry it took so long! looks good!

File: src/map/battle.cpp (hunk 1 above)
Comment (eppc0330) [reply src/map/battle.cpp:8801]: Oh right it sounds good. Thanks!

**Review by aleos89 — APPROVED** [review]
